src/docs/truffle/getting-started/interacting-with-your-contracts.md

```diff
@@ -110,119 +107,95 @@ There are three functions on the MetaCoin contract that we can execute. If you a
+<p class="alert alert-warning">
+**Warning**: We convert the return value to a number because in this example the numbers are small. However, if you try to convert a BigNumber that's larger than the largest integer supported by Javascript, you'll likely run into errors or unexpected behavior.
+</p>
 
-### Catching events
+### Processing transaction results
 
-Your contracts can fire events that you can catch to gain more insight into what your contracts are doing. The easiest way to handle events is by processing the result object of the transaction that triggered the event, like so:
+When you make a transaction, you're given a `result` object that gives you a wealth of information about the transaction. 
 
-```javascript
-var account_one = "0x1234..."; // an address
-var account_two = "0xabcd..."; // another address
-
-var meta;
-MetaCoin.deployed().then(function(instance) {
-  meta = instance;  
-  return meta.sendCoin(account_two, 10, {from: account_one});
-}).then(function(result) {
-  // result is an object with the following values:
-  //
-  // result.tx      => transaction hash, string
-  // result.logs    => array of decoded events that were triggered within this transaction
-  // result.receipt => transaction receipt object, which includes gas used
-
-  // We can loop through result.logs to see if we triggered the Transfer event.
-  for (var i = 0; i < result.logs.length; i++) {
-    var log = result.logs[i];
-
-    if (log.event == "Transfer") {
-      // We found the event!
-      break;
-    }
-  }
-}).catch(function(err) {
-  // There was an error! Handle it.
-});
+```shell
+truffle(develop)> let result = await contract.sendCoin(accounts[1], 10, {from: accounts[0]})
+truffle(develop)> result
 ```
 
-### Processing transaction results
-
-When you make a transaction, you're given a `result` object that gives you a wealth of information about the transaction. Specifically, you get the following:
+Specifically, you get the following:
 
 * `result.tx` *(string)* - Transaction hash
 * `result.logs` *(array)* - Decoded events (logs)
-* `result.receipt` *(object)* - Transaction receipt
+* `result.receipt` *(object)* - Transaction receipt (includes the amount of gas used)
 
 For more information, please see the [README](https://github.com/trufflesuite/truffle/tree/master/packages/truffle-contract) in the `truffle-contract` package.
 
+### Catching events
+
+Your contracts can fire events that you can catch to gain more insight into what your contracts are doing. The easiest way to handle events is by processing the `logs` array contained within `result` object of the transaction that triggered the event.
+
+If we explicitly output the first log entry we can see the details of the event that was emitted as part of the `sendCoin` call (`Transfer(msg.sender, receiver, amount);`).
+
+```shell
```

Comment:
(Just a note @kevinbluer: these should be `javascript` typically, just so things like the JSON-ish blob below format better. I know it's not really right.)